src/http/gateway/resources/gateway.js

```diff
@@ -153,7 +159,7 @@ module.exports = {
           log.error(err)
           return reject(err)
         }
-        resolve({ peekedStream, contentType: detectContentType(path, streamHead) })
+        resolve({ peekedStream, contentType: detectContentType(ipfsPath, streamHead) })
```

Comment:
`ipfsPath` instead of `path` because if we are returning implicit `index.html` we want to detect HTML content type of the file